### DISCUSSION TITLE
Image Should Take a WrapperClassname Prop

### DISCUSSION THREAD

**Discussion by machineghost** [vercel/next.js#21738] [Ideas]
**What version of Next.js are you using?**

10.0.5

**What version of Node.js are you using?**

15.4.0

**What browser are you using?**

Chrome

**What operating system are you using?**

Linux

**How are you deploying your application?**

This issue affects all environments, including dev (where I'm observing it)

**Describe the Bug**

It is impossible to style the HTML generated by the Image component.  Styles are expressly forbidden by the component, and `className` is recommended instead ... but `className` doesn't work, because it is applied to the (inner) `img` tag, not the outer/wrapping `div` that `Image` generates.

(As an aside: my kingdom for an `Image` component that generates an `img` DOM element!!! Or at least a better contained `div` + `img`.)

But if we can't get that, can we at least add a `wrapperClassName` prop, so that Next users have some way to style that div?  This would be in https://github.com/vercel/next.js/blob/356bcdec0302c13af312ddcc5c39e78ef8e40c9d/packages/next/client/image.tsx ... as you can see at line 421, the `className` prop is currently passed to the `img` ... but the wrapping `div` on 397 *only* gets a `style` prop (which the Next user can only sort of, indirectly, control).

Since this seems super simple, I'd be happy to submit a PR.

**Expected Behavior**

Next.js devs should have a way to style the *actual* element produced by the `Image` component ... not just a piece inside of it.

**To Reproduce**

Use an `Image` component, try to style the wrapper `div`, give up because it's impossible.

**Comment by Luisdv93** [comment]
This is definitely needed. The project I'm currently working on has a lot offset images where I need to play with negative margins to position but the overflow:hidden applied to the outer div makes it impossible to make it work unless I use another outer div.

**Comment by justintaddei** [comment]
This lack of an ability to style the wrapper `div` has caused me pain in every project I've tried to used it in. 99% of the time there is a need to apply some sort of styling an image, whether it be negative margins or something simple like a box-shadow (<- which is also currently impossible due to `overflow: hidden` being applied to the wrapper element).

I really hope this gets resolved soon. This could be such a powerful component if not for this one weakness.
